Integrations/Panorama/Panorama.py

```diff
@@ -2933,6 +2936,229 @@ def panorama_unregister_ip_tag_command():
     })
 
 
+''' Traffic Logs '''
+
+
+def build_traffic_logs_query(source=None, destination=None, receive_time=None, application=None, to_port=None, action=None):
+    query = ''
+    if source and len(source) > 0:
+        query += '(addr.src in ' + source + ')'
+    if destination and len(destination) > 0:
+        if len(query) > 0 and query[-1] == ')':
+            query += ' and '
+        query += '(addr.dst in ' + source + ')'
+    if receive_time and len(receive_time) > 0:
+        if len(query) > 0 and query[-1] == ')':
+            query += ' and '
+        query += '(receive_time geq ' + receive_time + ')'
+    if application and len(application) > 0:
+        if len(query) > 0 and query[-1] == ')':
+            query += ' and '
+        query += '(app eq ' + application + ')'
+    if to_port and len(to_port) > 0:
+        if len(query) > 0 and query[-1] == ')':
+            query += ' and '
+        query += '(port.dst eq ' + to_port + ')'
+    if action and len(action) > 0:
+        if len(query) > 0 and query[-1] == ')':
+            query += ' and '
+        query += '(action eq ' + action + ')'
+    return query
+
+
+@logger
+def panorama_query_traffic_logs(number_of_logs, direction, query,
+                                source, destination, receive_time, application, to_port, action):
+    params = {
+        'type': 'log',
+        'log-type': 'traffic',
+        'key': API_KEY
+    }
+
+    if query and len(query) > 0:
+        params['query'] = query
+    else:
+        params['query'] = build_traffic_logs_query(source, destination, receive_time, application, to_port, action)
+    if number_of_logs:
+        params['nlogs'] = number_of_logs
+    if direction:
+        params['dir'] = direction
+    result = http_request(
+        URL,
+        'GET',
+        params=params,
+    )
+
+    return result
+
+
+def panorama_query_traffic_logs_command():
+    """
+    Query the traffic logs
+    """
+    number_of_logs = demisto.args().get('number_of_logs')
+    direction = demisto.args().get('direction')
+    query = demisto.args().get('query')
+    source = demisto.args().get('source')
+    destination = demisto.args().get('destination')
+    receive_time = demisto.args().get('receive_time')
+    application = demisto.args().get('application')
+    to_port = demisto.args().get('to_port')
+    action = demisto.args().get('action')
+
+    if query and (source or destination or receive_time or application or to_port or action):
+        return_error('Use the query argument or the '
+                     'source, destination, receive_time, application, to_port, action arguments to build your query')
```

Comment:
I thin we need a better error message here that explains that you can only use one of them.
Like saying it is mutual exclusive. consult with @kirbles19

Reply:
Which message am I actually supposed to be reviewing?

Reply:
@kirbles19  lines 3011-3012.